packages/apps/spaces/app/organizations/[id]/components/Timeline/events/meeting/MeetingPreviewModal.tsx

```diff
@@ -0,0 +1,339 @@
+import { Tooltip } from '@ui/overlay/Tooltip';
+import { IconButton } from '@ui/form/IconButton';
+import { DateTimeUtils } from '@spaces/utils/date';
+import { ScaleFade } from '@ui/transitions/ScaleFade';
+import { ExternalSystemType, Meeting } from '@graphql/types';
+import { getGraphQLClient } from '@shared/util/getGraphQLClient';
+import { useCopyToClipboard } from '@spaces/hooks/useCopyToClipboard';
+import { Card, CardBody, CardHeader, CardFooter } from '@ui/presentation/Card';
+// import { FormAutoresizeTextarea } from '@ui/form/Textarea/FormAutoresizeTextarea';
+import { useUpdateMeetingMutation } from '@organization/graphql/updateMeeting.generated';
+
+import { useTimelineEventPreviewContext } from '../../preview/TimelineEventsPreviewContext/TimelineEventPreviewContext';
+import { MeetingIcon, HubspotIcon, CalcomIcon } from './icons';
+
+export const MeetingPreviewModal = () => {
+  const [_, copy] = useCopyToClipboard();
+  const client = getGraphQLClient();
+  const [isMounted, setIsMounted] = useState(false);
+  const updateMeeting = useUpdateMeetingMutation(client);
+  const { isModalOpen, closeModal, modalContent } =
+    useTimelineEventPreviewContext();
+
+  const event = modalContent as Meeting;
+  const creatorTimeZone =
+    event?.createdBy?.[0]?.__typename === 'ContactParticipant'
+      ? event?.createdBy?.[0]?.contactParticipant?.timezone
+      : '';
+
+  const creatorTimeZoneCode = creatorTimeZone
+    ? creatorTimeZone?.split(' ')?.[0]
+    : '';
+
+  const when = `${DateTimeUtils.format(
+    event?.startedAt,
+    DateTimeUtils.longWeekday,
+  )}, ${DateTimeUtils.format(
+    event?.startedAt,
+    DateTimeUtils.dateWithAbreviatedMonth,
+  )} • ${DateTimeUtils.format(
+    event.startedAt,
+    DateTimeUtils.usaTimeFormatString,
+  )} - ${DateTimeUtils.format(
+    event.endedAt,
+    DateTimeUtils.usaTimeFormatString,
+  )}`;
+
+  const zonedStartDateStr = creatorTimeZone
+    ? utcToZonedTime(event?.startedAt, creatorTimeZoneCode)
+    : null;
+  const zonedEndDateStr = creatorTimeZone
+    ? utcToZonedTime(event?.endedAt, creatorTimeZoneCode)
+    : null;
+
+  const zoned = (() => {
+    if (!zonedStartDateStr) return null;
+    const start = format(
+      zonedStartDateStr,
+      !zonedEndDateStr
+        ? DateTimeUtils.timeWithGMT
+        : DateTimeUtils.usaTimeFormatString,
+      {
+        timeZone: creatorTimeZoneCode || undefined,
+      },
+    );
+
+    if (!zonedEndDateStr) return start;
+
+    const end = format(zonedEndDateStr, DateTimeUtils.timeWithGMT, {
+      timeZone: creatorTimeZoneCode || undefined,
+    });
+
+    return `${start} - ${end}`;
+  })();
+
+  const owner = (() => {
+    if (event?.createdBy?.[0]?.__typename === 'ContactParticipant') {
+      const participant = event?.createdBy?.[0]?.contactParticipant;
+      return participant.emails?.[0]?.email ?? participant?.firstName;
+    }
+    return '';
+  })();
+
+  const participants = event?.attendedBy
```

Comment:
This does not include all possible types of participants. Please use helper function from timeline utils to ensure all possible types of participants are handled in consistent way